File: SampleLibrary/SampleLibrary/InvertedIndex.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

namespace SampleLibrary
{
    public class InvertedIndex
    {
        public static List<string> StopWords { get; } =
            FileReader.ReadFileContent("stopWords.txt").Split(',').ToList();

        public Dictionary<string, List<WordInfo>> Index { get; } = new Dictionary<string, List<WordInfo>>();

        public void IndexAllFiles(string folderAddress)
        {
            var listOfFiles = Directory.GetFiles(folderAddress);

            Console.WriteLine("indexing...");
            foreach (var fileAddress in listOfFiles) IndexFile(fileAddress);

            Console.WriteLine("indexing completed.");
        }

        private void IndexFile(string fileAddress)
        {
            var position = 0;
            var text = FileReader.ReadFileContent(fileAddress);
            foreach (var word in text.Split(' '))
            {
                var wordCopy = word.ToLower();

                position++;
                if (StopWords.Contains(wordCopy)) continue;

                if (!Index.ContainsKey(wordCopy)) Index.Add(wordCopy, new List<WordInfo>());

                Index[wordCopy].Add(new WordInfo(fileAddress, position));
            }
        }
    }
}

File: src/main/java/InvertedIndex.java
import java.io.File;
import java.util.*;

public class InvertedIndex {

    private static final List<String> stopWords;

    static {
        stopWords = Arrays.asList(FileReader.readFileContent(new File(FilePaths.STOP_WORDS.filePath)).split(","));
    }

    private Map<String, List<WordInfo>> index = new HashMap<>();

    public void indexAllFiles(String folderAddress) {
        File folder = new File(folderAddress);
        File[] listOfFiles = folder.listFiles();

        System.out.println("indexing...");
        for (File listOfFile : listOfFiles) {
            indexFile(new File(folderAddress + "/" + listOfFile.getName()));
        }
        System.out.println("indexing completed.");
    }

    private void indexFile(File file) {
        String fileName = file.getName();

        int position = 0;
        String text = FileReader.readFileContent(file);
        for (String word : text.split("\\W+")) {
            word = word.toLowerCase();
            position++;
            if (stopWords.contains(word))
                continue;
            List<WordInfo> idx = index.computeIfAbsent(word, k -> new LinkedList<>());
            idx.add(new WordInfo(fileName, position));
        }
    }

    public static List<String> getStopWords() {
        return stopWords;
    }

    public Map<String, List<WordInfo>> getIndex() {
        return index;
    }

}